UI/src/view/point_asset/massTransitStopForm.js

```diff
@@ -192,22 +230,30 @@
           wrapper = $('<div />').addClass('wrapper edit-mode');
         }
         streetViewHandler = getStreetView();
+
+        var components;
+        if (busStopTypeSelected == 7 && (applicationModel.getSelectedTool() == 'AddPointAsset') && selectedMassTransitStopModel.getCurrentAsset().id === undefined)  {
+          components = createNewServiceDropDown();
+        } else {
+          components = getAssetForm();
+        }
+
         wrapper.append(streetViewHandler.render())
-          .append($('<div />').addClass('form form-horizontal form-dark').attr('role', 'form').append(userInformationLog()).append(getAssetForm()));
+            .append($('<div />').addClass('form form-horizontal form-dark form-masstransitstop').attr('role', 'form').append(userInformationLog()).append(components));
 
-        var buttons = function(isTerminalBusStop) {
+        var buttons = function (busStopTypeSelected) {
           return $('<div/>').addClass('mass-transit-stop').addClass('form-controls')
-            .append(new ValidationErrorLabel().element)
-            .append(new SaveButton(isTerminalBusStop).element)
-            .append(new CancelButton().element);
+              .append(new ValidationErrorLabel().element)
+              .append(new SaveButton(busStopTypeSelected).element)
+              .append(new CancelButton().element);
         };
 
         function busStopHeader() {
           var header;
 
           if (_.isNumber(selectedMassTransitStopModel.getByProperty('nationalId'))) {
             header = $('<span>Valtakunnallinen ID: ' + selectedMassTransitStopModel.getByProperty('nationalId') + '</span>');
-          } else if (isTerminalBusStop) {
+          } else if (busStopTypeSelected == 6) {
```

Comment:
I understand the reason for the creation of busStopTypeSelected but make sense delete the method how validates if isTerminalBusStop or doesn't create a method to validate if is isServiceBusStop?

Because that, now we need to always perform the validations like busStopTypeSelected == 6 or busStopTypeSelected == 7, in the future if we modify the type of the bus stops we need to modify in all code instead of one place.

Reply:
@marquesfm Check modification on this code